hibernate-core/src/main/java/org/hibernate/jpa/internal/enhance/EnhancingClassTransformerImpl.java

```diff
@@ -23,9 +26,12 @@
 public class EnhancingClassTransformerImpl implements ClassTransformer {
 
 	private final EnhancementContext enhancementContext;
+	private final BytecodeProvider bytecodeProvider;
 
 	public EnhancingClassTransformerImpl(EnhancementContext enhancementContext) {
+		Objects.requireNonNull( enhancementContext );
 		this.enhancementContext = enhancementContext;
+		this.bytecodeProvider = BytecodeProviderInitiator.buildDefaultBytecodeProvider();
```

Comment:
Isn't that possibly introducing another problem? You won't be using the same bytecode provider as Hibernate ORM...

But I must admit I don't know what this class is used for anyway :/

Reply:
This is used by JPA's "enhance at deployment time" capability, used on EE servers.

And yes I'd like to make the BytecodeProvider implementation pluggable, but it really isn't today: choices are exclusively `bytebuddy` or `none`. Clearly this use case can't use `none`.

(there are other areas too in which it's implied that it needs to be the ByteBuddy one - I'd certainly like to improve that but I'm considering that a different issue)

Reply:
> And yes I'd like to make the BytecodeProvider implementation pluggable

I was thinking more that you're not using the same *instance* of `BytecodeProvider`, whereas before this patch, you were. I have no idea what kind of impact this can have, but since the bytecode provider is stateful, it might be a problem?